src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.SslProtocols.cs

```diff
@@ -155,7 +156,8 @@ public void DisabledProtocols_SetSslProtocols_ThrowsException(SslProtocols disab
                 }
                 using (var client = new HttpClient(handler))
                 {
-                    (await client.GetAsync(url)).Dispose();
+                    Type remoteServerExceptionType = RuntimeInformation.IsOSPlatform(OSPlatform.Linux) ? typeof(TaskCanceledException) : typeof(HttpRequestException);
```

Comment:
Why do we think Linux only returns TaskCanceledException and non-Linux returns HttpRequestExceptions?

Any of those exceptions could be returned from either platform.

Reply:
On linux, the libcurl connect timeout is set to int.maxvalue as default, whereas httpclient has default timout as 100s, so on linux the httpclient is getting cancelled with a cancellationtoken.

Reply:
>On linux, the libcurl connect timeout is set to int.maxvalue as default, whereas httpclient has default timout as 100s, so on linux the httpclient is getting cancelled with a cancellationtoken. 

int.maxvalue for a connect timeout should be changed.  It should be more like 30 seconds or something which is what all the other platforms like .NET Framework etc. use for TCP connection timeout.

Reply:
filed issue #25883